_posts/2020-11-04-oneway-sslv2hello.adoc

```diff
@@ -0,0 +1,153 @@
+
+What is your first and last name?
+   [Unknown]:  localhost
+What is the name of your organizational unit?
+   [Unknown]:  wildfly
+What is the name of your organization?
+   [Unknown]:  jboss
+What is the name of your City or Locality?
+   [Unknown]:  Raleigh
+What is the name of your State or Province?
+   [Unknown]:  Carolina
+What is the two-letter country code for this unit?
+   [Unknown]:  US
+Is CN=localhost, OU=wildfly, O=jboss, L=Raleigh, ST=Carolina, C=US correct?
+   [no]:  yes
+----
+
+Make sure you enter your desired "hostname" for the ``first and last name`` field, otherwise you might
+run into issues while permanently accepting this certificate
+as an exception to some browsers. Chrome does not currently exhibit this behaviour.
+
+=== Configuring the Server
+
+You configure the SSL context by running JBoss CLI commands. For your convenience, the example
+application batches commands into a ``configure-ssl.cli`` script provided
+in its root directory.
+
+Before you begin, make sure you start the server with the standalone default profile as follows:
+
+[source,shell]
+----
+WILDFLY_HOME/bin/standalone.sh
+----
+
+Then, run the script which configures your server SSL context to accept ``SSLv2Hello`` as one of its
+enabled protocols.
+
+[source,shell]
+----
+$ WILDFLY_HOME/bin/jboss-cli.sh --connect --file=configure-ssl.cli
+----
+
+=== Test the Server SSL Configuration
+To test the connection to the SSL port of your server instance, open a browser and navigate to
+https://localhost:8443/.
+
+NOTE: You get the privacy error because the
+server certificate is self-signed. If you ned to use a fully signed certificate, you must
+get a PEM file from the Certificate Authority and then import the PEM into the keystore.
+
+At this point in time, your server has been configured to accept ``SSLv2Hello`` messages,
+however we are yet to test whether a connection using SSLv2Hello would be successful,
+as your browser is not sending ``SSLv2Hello`` messages.
+
+=== Configure a Client that sends SSLv2Hello messages
+
+The example application presents a simple client ``Client.java`` which checks the connection
+to the server is successful. To ensure the client uses ``SSLv2Hello`` in its initial
+handshake, we have provided a common configuration framework in
+``wildfly-config.xml``.
+
+Review this file to see how ``SSLv2Hello`` is enabled in the client SSL context along with
+``TLSv1``.
+
+NOTE: When configuring any SSL context to enable ``SSLv2Hello`` (client or server), you always
+need to specify an additional SSL/TLS protocol. ``SSLv2Hello`` cannot be specified by itself,
+as it is a pseudo protocol which is only used to negotiate another SSL/TLS version to be used
+throughout the rest of the communication.
+
+As a final step in configuring the client, update the path to the client's trustore to
+``WILDFLY_HOME/standalone/configuration/server.keystore``.
```

Comment:
The client configuration should be making use of its own trust store. To create this, the server's certificate will need to be exported to a file (the `key-store` `export-certificate` operation can be used to do this) and then `keytool` can be used to import that certificate into the client's trust store.